fabric-transactions-v1/src/main/java/net/fabricmc/fabric/impl/transaction/TransactionInit.java

```diff
@@ -0,0 +1,13 @@
+package net.fabricmc.fabric.impl.transaction;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.event.server.ServerStartCallback;
+
+public class TransactionInit implements ModInitializer {
+	@Override
+	public void onInitialize() {
+		ServerStartCallback.EVENT.register(s -> {
+			TransactionImpl.setServerThread(s);
```

Comment:
We should unbind this on server shutdown such as for integrated servers.

Reply:
You mean set the thread reference to null? That does makes sense. Not sure what else to unbind.